packages/pluggableWidgets/video-player-native/src/VideoPlayer.tsx

```diff
@@ -36,27 +53,133 @@ export function VideoPlayer(props: VideoPlayerProps<VideoStyle>): ReactElement {
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [fullScreen]);
+
+    function handleShowControls(setShown?: boolean): void {
+        if (!props.showControls) {
+            return;
+        }
+
+        if (timeoutRef.current) {
+            clearTimeout(timeoutRef.current as NodeJS.Timeout);
+        }
+
+        if (!showControls || setShown) {
+            setShowControls(true);
+            timeoutRef.current = setTimeout(() => {
+                setShowControls(false);
+            }, 5000);
+        } else {
+            setShowControls(false);
+        }
+    }
+
+    function fullScreenHandler(isFullScreen: boolean): void {
+        setFullScreen(isFullScreen);
+        StatusBar.setHidden(isFullScreen);
+    }
+
+    const videoProps: VideoProperties = {
+        testID: props.name,
+        source: { uri: isAvailable(props.videoUrl) ? props.videoUrl.value : undefined },
+        muted: props.muted,
+        repeat: props.loop,
+        controls: props.showControls,
+        onLoadStart: () => setStatus(StatusEnum.LOADING),
+        onError: () => {
+            setStatus(StatusEnum.ERROR);
+        },
+        useTextureView: false,
+        resizeMode: props.aspectRatio ? "contain" : "stretch",
+        onProgress: ({ currentTime }: OnProgressData) => {
+            setCurrentPlayTime(currentTime);
+        }
+    };
+
+    const isAndroid = Platform.OS === "android";
+
     return (
-        <View style={styles.container}>
-            {status === StatusEnum.LOADING && <ActivityIndicator color={styles.indicator.color} size="large" />}
-            {status === StatusEnum.ERROR && <Text style={styles.errorMessage}>The video failed to load :(</Text>}
-            <Video
-                testID={props.name}
-                source={{ uri: isAvailable(props.videoUrl) ? props.videoUrl.value : undefined }}
-                paused={!props.autoStart}
-                muted={props.muted}
-                repeat={props.loop}
-                controls={props.showControls}
-                onLoadStart={() => setStatus(StatusEnum.LOADING)}
-                onLoad={data => {
-                    setStatus(StatusEnum.READY);
-                    setVideoAspectRatio(data.naturalSize.width / data.naturalSize.height);
-                }}
-                onError={() => setStatus(StatusEnum.ERROR)}
-                style={status !== StatusEnum.READY ? { height: 0 } : styles.video}
-                useTextureView={false}
-                resizeMode={props.aspectRatio ? "contain" : "stretch"}
-            />
-        </View>
+        <Fragment>
+            {isAndroid && (
+                <Modal
+                    isVisible={fullScreen}
+                    style={styles.fullScreenVideoPlayer}
+                    onBackButtonPress={() => {
+                        fullScreenHandler(false);
+                    }}
+                >
+                    <TouchableWithoutFeedback onPress={() => handleShowControls()}>
```

Comment:
Maybe this is better `onPress={props.showControls && handleShowControls}` and then you can remove lines https://github.com/mendix/widgets-resources/pull/1303/files#diff-5b19cf908530b18852147d085549427ea128e2eb1da3dddad52826adb3d6aae7R64-R66

Remember to do this elsewhere

Reply:
this is not possible because this can return boolean which is not an allowed type, then I will have to write it this way  `props.showControls ? handleShowControls: undefined` and I will have to it twice. I prefer to add it to the function body to avoid code duplication.

Reply:
Okay